my_work_R/my_happy_script.R

```diff
@@ -1,93 +1,101 @@
-par(omi = c(0,0,0.05,0.05), mai = c(1,1,0,0), las = 1, mgp = c(2,.5,0), cex = 1.5)
-
-plot(NA, NA, xlim = c(2, 1000), ylim = c(4.7, 0.75),
-     ylab = "Test RMSE (°C)", xlab = "Training temperature profiles (#)", log = 'x', axes = FALSE)
-
-n_profs <- c(2, 10, 50, 100, 500, 980)
-
-axis(1, at = c(-100, n_profs, 1e10), labels = c("", n_profs, ""), tck = -0.01)
-axis(2, at = seq(0,10), las = 1, tck = -0.01)
-
-# slight horizontal offsets so the markers don't overlap:
-offsets <- data.frame(pgdl = c(0.15, 0.5, 3, 7, 20, 30)) %>%
-  mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
-
-
-for (mod in c('pb','dl','pgdl')){
-  mod_data <- filter(eval_data, model_type == mod)
-  mod_profiles <- unique(mod_data$n_prof)
-  for (mod_profile in mod_profiles){
-    d <- filter(mod_data, n_prof == mod_profile) %>% summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
-    x_pos <- offsets %>% filter(n_prof == mod_profile) %>% pull(!!mod) + mod_profile
-    lines(c(x_pos, x_pos), c(d$y0, d$y1), col = d$col, lwd = 2.5)
-  }
-  d <- group_by(mod_data, n_prof) %>% summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
-    rename(x = n_prof) %>% arrange(x)
-  
-  lines(d$x + tail(offsets[[mod]], nrow(d)), d$y, col = d$col[1], lty = 'dashed')
-  points(d$x + tail(offsets[[mod]], nrow(d)), d$y, pch = d$pch[1], col = d$col[1], bg = 'white', lwd = 2.5, cex = 1.5)
-  
-}
-
-points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 0.80, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
-
-points(2.2, 0.94, col = '#d95f02', pch = 22, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 0.95, 'Deep Learning', pos = 4, cex = 1.1)
-
-points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
-
-dev.off()
+eval_data <- readr::read_csv(mendota_file, col_types = 'iccd') |>
+  filter(str_detect(exper_id, 'similar_[0-9]+')) |>
+  group_by(exper_id, model_type) |> 
+  summarize(
+    mean_rmse = mean(rmse),
+    min_rmse = min(rmse),
+    max_rmse = max(rmse),
+    ct_exper = n()
+  )
+
+eval_data <- eval_data |> 
+  mutate(
+    col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+      ), 
+    pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+      ), 
+    n_prof = as.numeric(str_extract(exper_id, '[0-9]+')),
+    model_type = factor(model_type, levels = c("pgdl", "dl", "pb"))
+    ) |> 
+    mutate(
+      model_legend = fct_recode(
+        model_type,
+        "Process-Guided Deep Learning" = "pgdl",
+        "Deep Learning" = "dl",
+        "Process-Based" = "pb"
+      )
+    )
+
+g <- 
+  ggplot(data = eval_data, 
+         aes(color = model_legend, shape = model_legend)) +
+  geom_line(aes(x = n_prof, y = mean_rmse), 
+            linetype = "dashed",
```

Comment:
I think the prior was "dotted" but either works

Reply:
resolved